test_unit.py

```diff
@@ -36,3 +35,45 @@ def test_host_fips_disabled(tmp_path):
     fipsfile = tmp_path / "fips"
     fipsfile.write_text("")
     assert not host_fips_enabled(f"{fipsfile}")
+
+
+def test_routing_keys():
+    con = RabbitMQConnection(message_topic_prefix="foo.bar")
+    assert con.repository_routing_keys == [
+        "foo.bar." + s for s in ("repo.publish_state", "repo.published")
+    ]
+
+
+def test_container_list():
+    con = RabbitMQConnection(
+        container_list=[
+            Container(
+                type="openjdk-devel",
+                repo="devel/bci/images",
+                image="bci/openjdk-devel",
+                tag="16",
+                version="16",
+            ),
+            Container(
+                type="python",
+                repo="home/fcrozat/matryoshka/containers_python36",
+                image="python36",
+                tag="latest",
+                version="3.6",
+            ),
+            Container(
+                type="node",
+                repo="home/fcrozat/matryoshka/containers_node15",
+                image="python15",
+                tag="latest",
+                version="15",
+            ),
+        ]
+    )
+    print(con.watched_projects)
+    assert con.watched_projects["devel:bci"] == ["images"]
```

Comment:
That will probably break soon, but I expect it's a good thing?

Reply:
No, it won't break, because this is calculated from the list in line 49.